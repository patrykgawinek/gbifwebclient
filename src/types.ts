export type ClassificationState = {
  name: string;
  value: number;
  setValue: React.Dispatch<React.SetStateAction<number>>;
};

/* Countries type */
export type Country = {
  Code: string;
  Name: string;
};

/* Species type */
export type Classification = {
  key: number;
  nubKey: number;
  nameKey: number;
  taxonID: string;
  sourceTaxonKey: number;
  kingdom: string;
  phylum: string;
  kingdomKey: number;
  phylumKey: number;
  datasetKey: string;
  constituentKey: string;
  parentKey: number;
  parent: string;
  scientificName: string;
  canonicalName: string;
  vernacularName: string;
  authorship: string;
  nameType: string;
  rank: string;
  origin: string;
  taxonomicStatus: string;
  nomenclaturalStatus: any[];
  remarks: string;
  numDescendants: number;
  lastCrawled: Date;
  lastInterpreted: Date;
  issues: string[];
  synonym: boolean;
  publishedIn: string;
  family: string;
  familyKey?: number;
};

export type Species = {
  offset: number;
  limit: number;
  endOfRecords: boolean;
  results: Classification[];
};

/* Occurence results type */
export type HttpRsGbifOrgTerms10Multimedia = {
  'http://purl.org/dc/terms/title': string;
  'http://purl.org/dc/terms/format': string;
  'http://purl.org/dc/terms/license': string;
  'http://purl.org/dc/terms/references': string;
  'http://purl.org/dc/terms/creator': string;
  'http://purl.org/dc/terms/rightsHolder': string;
  'http://purl.org/dc/terms/identifier': string;
};

export type Extensions = {
  'http://rs.gbif.org/terms/1.0/Multimedia'?: HttpRsGbifOrgTerms10Multimedia[];
};

export type Medium = {
  type: string;
  format: string;
  identifier: string;
  references: string;
  title: string;
  creator: string;
  license: string;
  rightsHolder: string;
};

export type Level0 = {
  gid: string;
  name: string;
};

export type Level1 = {
  gid: string;
  name: string;
};

export type Level2 = {
  gid: string;
  name: string;
};

export type Level3 = {
  gid: string;
  name: string;
};

export type Gadm = {
  level0?: Level0;
  level1?: Level1;
  level2?: Level2;
  level3?: Level3;
};

export type Occurence = {
  key: number;
  datasetKey: string;
  publishingOrgKey: string;
  networkKeys: string[];
  installationKey: string;
  publishingCountry: string;
  protocol: string;
  lastCrawled: Date;
  lastParsed: Date;
  crawlId: number;
  hostingOrganizationKey: string;
  extensions: Extensions;
  basisOfRecord: string;
  occurrenceStatus: string;
  lifeStage: string;
  taxonKey: number;
  kingdomKey: number;
  phylumKey: number;
  classKey: number;
  orderKey: number;
  familyKey: number;
  genusKey: number;
  speciesKey: number;
  acceptedTaxonKey: number;
  scientificName: string;
  acceptedScientificName: string;
  kingdom?: string;
  phylum?: string;
  order?: string;
  family?: string;
  genus?: string;
  species?: string;
  genericName: string;
  specificEpithet: string;
  taxonRank: string;
  taxonomicStatus: string;
  decimalLongitude: number;
  decimalLatitude: number;
  coordinatePrecision: number;
  coordinateUncertaintyInMeters: number;
  stateProvince: string;
  year: number;
  month: number;
  day: number;
  eventDate: Date;
  issues: string[];
  lastInterpreted: Date;
  license: string;
  identifiers: any[];
  media: Medium[];
  facts: any[];
  relations: any[];
  gadm: Gadm;
  institutionKey: string;
  isInCluster: boolean;
  geodeticDatum: string;
  class: string;
  countryCode: string;
  recordedByIDs: any[];
  identifiedByIDs: any[];
  country: string;
  identifier: string;
  eventID: string;
  'http://unknown.org/taxonRankID': string;
  dynamicProperties: string;
  collectionCode: string;
  verbatimLocality: string;
  gbifID: string;
  occurrenceID: string;
  'http://unknown.org/subspecies': string;
  catalogNumber: string;
  recordedBy: string;
  vernacularName: string;
  otherCatalogNumbers: string;
  institutionCode: string;
  taxonConceptID: string;
  'http://unknown.org/species': string;
  identifiedBy: string;
  individualCount?: number;
  associatedReferences: string;
  county: string;
  locality: string;
  municipality: string;
  verbatimElevation: string;
  iucnRedListCategory: string;
  waterBody: string;
  verbatimEventDate: string;
  datasetName: string;
  eventRemarks: string;
  language: string;
  type: string;
  occurrenceRemarks: string;
  projectId: string;
  establishmentMeans: string;
  rightsHolder: string;
  dataGeneralizations: string;
  verbatimCoordinateSystem: string;
  identificationVerificationStatus: string;
  dateIdentified?: Date;
  continent: string;
  modified?: Date;
  fieldNumber: string;
  earliestEraOrLowestErathem: string;
  earliestEpochOrLowestSeries: string;
  preparations: string;
  lowestBiostratigraphicZone: string;
  locationID: string;
  earliestPeriodOrLowestSystem: string;
  group: string;
  elevation?: number;
  habitat: string;
  nomenclaturalCode: string;
  ownerInstitutionCode: string;
  samplingProtocol: string;
  higherClassification: string;
  fieldNotes: string;
  eventTime: string;
  behavior: string;
  identificationQualifier: string;
  'http://unknown.org/rightsHolder': string;
  'http://unknown.org/license': string;
  taxonID: string;
  datasetID: string;
  references: string;
  organismQuantity?: number;
  organismQuantityType: string;
  'http://rs.tdwg.org/dwc/terms/organismQuantity': string;
  organismName: string;
  'http://rs.tdwg.org/dwc/terms/organismQuantityType': string;
};

export type Occurences = {
  offset: number;
  limit: number;
  endOfRecords: boolean;
  count: number;
  results: Occurence[];
  facets: any[];
};
