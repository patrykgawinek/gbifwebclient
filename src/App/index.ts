export { App, Theme } from "./App";
