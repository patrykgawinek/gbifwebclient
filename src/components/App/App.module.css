.AppContainer {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.mainContent {
  margin: 0 auto;
  max-width: 1140px;
}
