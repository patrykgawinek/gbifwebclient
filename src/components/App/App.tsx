import HomePage from "pages/HomePage/HomePage";
import SingleOccurencePage from "pages/SingleOccurencePage/SingleOccurencePage";
import React from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Header from "../Header/Header";
import "./App.module.css";

function App() {
  let id: number = 5;
  return (
    <div>
      <Router>
        <Header />
        <Switch>
          <Route path="/">
            <HomePage />
          </Route>
          <Route path="/occurences/:id">
            <SingleOccurencePage id={id} />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
