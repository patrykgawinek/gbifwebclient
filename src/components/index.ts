export { SearchClassification, SearchKingdom } from './SearchOccurrences';
export { SearchOccurrencesResults, SingleOccurrenceResult } from './SearchOccurrencesResult';
export { SelectCountry } from './SelectCountry';
