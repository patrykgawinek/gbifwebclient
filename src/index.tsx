import React from "react";
import { createRoot } from "react-dom/client";
import { App } from "./App";

import "./index.css";
import "bootstrap/dist/css/bootstrap.min.css";

const container = document.getElementById("root") as HTMLElement;
const root = createRoot(container);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
);
