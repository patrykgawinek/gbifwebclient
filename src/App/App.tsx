import { createContext, useEffect, useState } from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";

import styles from "./App.module.css";

import { HomePage } from "../pages/HomePage";
import { MapPage } from "../pages/MapPage";
import { SearchOccurrencesPage } from "../pages/SearchOccurrencesPage";
import { SingleOccurrencePage } from "../pages/SingleOccurrencePage";

import { Footer } from "../components/Footer";
import { Header } from "../components/Header";

export const Theme = createContext({
  darkMode: true,
  setDarkMode: (_mode: boolean) => {},
});

export const App: React.FC = () => {
  const [darkMode, setDarkMode] = useState<boolean>(true);

  useEffect(() => {
    let data = localStorage.getItem("theme");
    if (data === "false") {
      setDarkMode(false);
    }
  }, []);

  useEffect(() => {
    localStorage.setItem("theme", JSON.stringify(darkMode));
  }, [darkMode]);

  return (
    <Theme.Provider value={{ darkMode: darkMode, setDarkMode: setDarkMode }}>
      <Router>
        <div className={`${styles.AppContainer} ${darkMode ? styles.darkMode : undefined}`}>
          <Header />
          <Routes>
            <Route path="/">
              <Route
                index
                element={
                  <div className={styles.mainContent}>
                    <HomePage />
                  </div>
                }
              />
              <Route path="occurrences">
                <Route
                  index
                  element={
                    <div className={styles.mainContent}>
                      <SearchOccurrencesPage />
                    </div>
                  }
                />
                <Route
                  path=":id"
                  element={
                    <div className={styles.mainContent}>
                      <SingleOccurrencePage />
                    </div>
                  }
                />
              </Route>
              <Route path="map" element={<MapPage />} />
            </Route>
          </Routes>
          <Footer />
        </div>
      </Router>
    </Theme.Provider>
  );
};
