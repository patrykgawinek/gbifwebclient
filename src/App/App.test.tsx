import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { App } from "src/App";
import React from "react";

describe("App root", () => {
  test("renders home page", () => {
    const component: React.ReactElement = <App />;
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
  });
});
