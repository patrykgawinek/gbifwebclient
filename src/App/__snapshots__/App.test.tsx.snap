// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App root > renders home page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_AppContainer_8c7554 _darkMode_8c7554"
      >
        <header>
          <nav
            class="bg-dark navbar-dark navbar navbar-expand-sm navbar-light"
          >
            <div
              class="container"
            >
              <a
                aria-current="page"
                class="_brand_a5b977 active"
                data-discover="true"
                href="/"
              >
                <span
                  class="navbar-brand"
                >
                  GBIF Client
                </span>
              </a>
              <button
                aria-controls="basic-navbar-nav"
                aria-label="Toggle navigation"
                class="navbar-toggler collapsed"
                type="button"
              >
                <span
                  class="navbar-toggler-icon"
                />
              </button>
              <div
                class="navbar-collapse collapse"
                id="basic-navbar-nav"
              >
                <div
                  class="me-auto navbar-nav"
                >
                  <a
                    aria-current="page"
                    class="nav-link active"
                    data-discover="true"
                    href="/"
                  >
                    How-to
                  </a>
                  <a
                    class="nav-link"
                    data-discover="true"
                    href="/occurrences"
                  >
                    Search
                  </a>
                  <a
                    class="nav-link"
                    data-discover="true"
                    href="/map"
                  >
                    Map
                  </a>
                </div>
                <button
                  class="_themeToggle_a5b977 btn-dark btn btn-primary"
                  type="button"
                >
                  <img
                    alt="Dark mode"
                    class="_modeButtonIcon_a5b977"
                    src="/assets/icons/moon.png"
                  />
                </button>
              </div>
            </div>
          </nav>
        </header>
        <div
          class="_mainContent_8c7554"
        >
          <main>
            <div
              class="container"
            >
              <h1
                class="mt-3 _lightText_7e21aa"
              >
                What is the GBIF Client web app?
              </h1>
              <p
                class="_lightText_7e21aa"
              >
                The GBIF Client web app is a front-end web app that gives you a more user friendly experience when accessing
                 
                <a
                  class="_hyperLinkDark_7e21aa"
                  href="https://www.gbif.org/occurrence/search"
                >
                  GBIF's
                </a>
                 
                dataset. GBIF is an international network and data infrastructure funded by the world's governments and aimed at providing anyone, anywhere, open access to data about all types of life on Earth. This means that you can find data on a plethora of species and recorded occurrences of said species.
              </p>
              <h2
                class="_lightText_7e21aa"
              >
                How do I use this web app?
              </h2>
              <p
                class="_lightText_7e21aa"
              >
                To use this web app, you can simply navigate to the
                 
                <a
                  class="_hyperLinkDark_7e21aa"
                  data-discover="true"
                  href="/occurrences"
                >
                  search page
                </a>
                 
                and start browsing through different kingdoms, phyla ... Below the search section found occurrences will appear. You can select an occurrence to see more details, or you can click the button to show all occurrences on a heatmap.
              </p>
              <h2
                class="_lightText_7e21aa"
              >
                What is the difference between the map button on the search page and the map in the top navigation?
              </h2>
              <p
                class="_lightText_7e21aa"
              >
                The button on the search page shows a heatmap for the taxonomy you selected. The
                 
                <a
                  class="_hyperLinkDark_7e21aa"
                  data-discover="true"
                  href="/map"
                >
                  map in the navigation
                </a>
                , however, shows a heatmap of all occurrences in the dataset of GBIF.
              </p>
              <h2
                class="_lightText_7e21aa"
              >
                Does this web app save any of my data?
              </h2>
              <p
                class="_lightText_7e21aa"
              >
                We save your choice of theme locally on your computer. You can always remove said item by clearing your browser's cache. Alongside of that we save your last selected taxonomy. This gets cleared every time you close the tab this web app was opened with.
              </p>
              <button
                class="mb-3 _buttonStyle_7e21aa btn btn-dark"
                type="button"
              >
                Start browsing!
              </button>
            </div>
          </main>
        </div>
        <footer>
          <nav
            class="bg-dark navbar-dark navbar navbar-expand navbar-light"
          >
            <p
              class="_footerText_acfd2c"
            >
              GBIF Client | Â©
              2025
               | Project for AP Hogeschool - Patryk Gawinek
               
            </p>
          </nav>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_AppContainer_8c7554 _darkMode_8c7554"
    >
      <header>
        <nav
          class="bg-dark navbar-dark navbar navbar-expand-sm navbar-light"
        >
          <div
            class="container"
          >
            <a
              aria-current="page"
              class="_brand_a5b977 active"
              data-discover="true"
              href="/"
            >
              <span
                class="navbar-brand"
              >
                GBIF Client
              </span>
            </a>
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <div
                class="me-auto navbar-nav"
              >
                <a
                  aria-current="page"
                  class="nav-link active"
                  data-discover="true"
                  href="/"
                >
                  How-to
                </a>
                <a
                  class="nav-link"
                  data-discover="true"
                  href="/occurrences"
                >
                  Search
                </a>
                <a
                  class="nav-link"
                  data-discover="true"
                  href="/map"
                >
                  Map
                </a>
              </div>
              <button
                class="_themeToggle_a5b977 btn-dark btn btn-primary"
                type="button"
              >
                <img
                  alt="Dark mode"
                  class="_modeButtonIcon_a5b977"
                  src="/assets/icons/moon.png"
                />
              </button>
            </div>
          </div>
        </nav>
      </header>
      <div
        class="_mainContent_8c7554"
      >
        <main>
          <div
            class="container"
          >
            <h1
              class="mt-3 _lightText_7e21aa"
            >
              What is the GBIF Client web app?
            </h1>
            <p
              class="_lightText_7e21aa"
            >
              The GBIF Client web app is a front-end web app that gives you a more user friendly experience when accessing
               
              <a
                class="_hyperLinkDark_7e21aa"
                href="https://www.gbif.org/occurrence/search"
              >
                GBIF's
              </a>
               
              dataset. GBIF is an international network and data infrastructure funded by the world's governments and aimed at providing anyone, anywhere, open access to data about all types of life on Earth. This means that you can find data on a plethora of species and recorded occurrences of said species.
            </p>
            <h2
              class="_lightText_7e21aa"
            >
              How do I use this web app?
            </h2>
            <p
              class="_lightText_7e21aa"
            >
              To use this web app, you can simply navigate to the
               
              <a
                class="_hyperLinkDark_7e21aa"
                data-discover="true"
                href="/occurrences"
              >
                search page
              </a>
               
              and start browsing through different kingdoms, phyla ... Below the search section found occurrences will appear. You can select an occurrence to see more details, or you can click the button to show all occurrences on a heatmap.
            </p>
            <h2
              class="_lightText_7e21aa"
            >
              What is the difference between the map button on the search page and the map in the top navigation?
            </h2>
            <p
              class="_lightText_7e21aa"
            >
              The button on the search page shows a heatmap for the taxonomy you selected. The
               
              <a
                class="_hyperLinkDark_7e21aa"
                data-discover="true"
                href="/map"
              >
                map in the navigation
              </a>
              , however, shows a heatmap of all occurrences in the dataset of GBIF.
            </p>
            <h2
              class="_lightText_7e21aa"
            >
              Does this web app save any of my data?
            </h2>
            <p
              class="_lightText_7e21aa"
            >
              We save your choice of theme locally on your computer. You can always remove said item by clearing your browser's cache. Alongside of that we save your last selected taxonomy. This gets cleared every time you close the tab this web app was opened with.
            </p>
            <button
              class="mb-3 _buttonStyle_7e21aa btn btn-dark"
              type="button"
            >
              Start browsing!
            </button>
          </div>
        </main>
      </div>
      <footer>
        <nav
          class="bg-dark navbar-dark navbar navbar-expand navbar-light"
        >
          <p
            class="_footerText_acfd2c"
          >
            GBIF Client | Â©
            2025
             | Project for AP Hogeschool - Patryk Gawinek
             
          </p>
        </nav>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
